applications/system-service/systemapi.h

```diff
@@ -211,6 +210,65 @@ class SystemAPI : public APIBase {
         toggleSwipeEnabled((SwipeDirection)direction);
     }
     void toggleSwipeEnabled(SwipeDirection direction){ setSwipeEnabled(direction, !getSwipeEnabled(direction)); }
+    Q_INVOKABLE void setSwipeLength(int direction, int length){
+        int maxLength;
+        if(!hasPermission("system")){
+            return;
+        }
+        if(direction <= SwipeDirection::None || direction > SwipeDirection::Down){
+            qDebug() << "Invalid swipe direction: " << direction;
+            return;
+        }
+        if (direction == SwipeDirection::Up || direction == SwipeDirection::Down){
+            maxLength = 1872;
```

Comment:
This should be pulled from device settings as it could be different per device, and in the future I'll likely be pulling it dynamically.
```suggestion
            maxLength = deviceSettings->getTouchHeight();
```